src/services/codefixes/convertToAsyncFunction.ts

```diff
@@ -2,11 +2,13 @@
 namespace ts.codefix {
     const fixId = "convertToAsyncFunction";
     const errorCodes = [Diagnostics.This_may_be_converted_to_an_async_function.code];
+    let codeActionSucceeded = true;

```

Comment:
>codeActionSucceeded [](start = 8, length = 19)

I assume this is the thing you were talking about when you asked about exceptions.  It's not what I would have done, but it sounds like you've already checked the local conventions.  I might change it to `codeActionFailed` though.  As long as we're going down this path, would it make sense to have an error code indicating how it went wrong and then send a telemetry event when we see a failure?

Reply:
I'm not sure this is an appropriate place to bubble up an error code given the change to be more conservative about offering the diagnostic. That is, we won't show the diagnostic if we know the code fix is going to fail.